mmpose/core/post_processing/matching.py

```diff
@@ -0,0 +1,289 @@
+import numpy as np
+import torch
+from scipy.optimize import linear_sum_assignment
+
+sigmas = np.array([
+    .26, .25, .25, .35, .35, .79, .79, .72, .72, .62, .62, 1.07, 1.07, .87,
+    .87, .89, .89
+])
+
+
+def candidate_reselect(bboxes, pose_preds, num_joints, img, box_scores,
+                       in_vis_thre):
+    """get final result with group and match.
+
+    Note:
+        num_person: N
+    Args:
+        bboxes (torch.Tensor[N,4]): bbox(x1,y1,x2,y2).
+        pose_preds (dict): candidate keypoints information.
+            pose_preds[i][j] (list): i-th person j-th joint 's canditate
+                    keypoints information.
+        num_joints (int): number of keypoints.
+        img (int): image_id.
+        box_scores (torch.Tensor[N,1]): bbox score.
+        in_vis_thre (float): threshold.
+    Returns:
+        final_result (list): predicted in the image.
+    """
+
+    # Group same keypointns together
+    kp_groups = _grouping(bboxes, pose_preds, num_joints, box_scores)
+
+    # Generate Matrix
+    num_person = len(pose_preds.keys())
+    costMatrix = [
+        np.zeros((num_person, len(kp_group)), dtype=np.float)
+        for kp_group in kp_groups
+    ]
+
+    for h_id, person in pose_preds.items():
+        assert 0 <= h_id < num_person
+        for k in range(num_joints):
+            g_id = person['group_id'][k]
+            if g_id is not None:
+                # because group_id start with 1
+                g_id = int(g_id) - 1
+                _, _, score = person[k][0]
+                h_score = person['person_score']
+
+                if score < 0.05:
```

Comment:
Maybe we should add a new arg to this function: `score_thr = 0.05`

Reply:
agree

Reply:
What is the difference between `vis_thr`  and `score_thr (0.05 here)` ? @xiaohangcd @Canwang-sjtu 
Do we really need so many super-parameters?